K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -219,32 +220,30 @@ public static void AddAggregates(this AggregateDictionary aggregateDictionary, s
                 }
 
                 // Column Metadata (Separator %)
-                // Structure: key%metric%value1%value2%keyed
                 var columnMetadata = column.ColumnName.Split(AggregationsConstants.MetadataSeparator);
+                string key;
 
-                if (columnMetadata.Length > 1)
+                if (columnMetadata.Length <= 1)
                 {
-                    var metric = columnMetadata[1];
-
-                    if (metric == "percentile")
-                    {
-                        var key = columnMetadata[0];
-                        aggregateDictionary.Add(key, GetPercentileAggregate(column.ColumnName, columnMetadata, row, logger));
-                    }
-                    else if (metric == "extended_stats")
-                    {
-                        var key = columnMetadata[0];
-                        aggregateDictionary.Add(key, GetExtendedStatsAggregate(column.ColumnName, columnMetadata, row, logger));
-                    }
-                    else
-                    {
-                        throw new InvalidOperationException($"Failed to parse column metadata. {metric} is invalid.");
-                    }
+                    // ColumnName contains only the key (this is a value aggregate)
+                    key = column.ColumnName;
```

Comment:
Use early continue to cut down on indentation:

```c#
                if (columnMetadata.Length <= 1)
                {
                    // ColumnName contains only the key (this is a value aggregate)
                    key = column.ColumnName;
                    aggregateDictionary.Add(key, GetValueAggregate(key, row, logger));
                    continue;
                }

                // ColumnName contains key + metadata (metric used to select aggregate type)
                key = columnMetadata[0];
                var metric = columnMetadata[1];

                IAggregate aggregate = metric switch
                {
                    AggregationsConstants.Percentile => GetPercentileAggregate(column.ColumnName, columnMetadata, row, logger),
                    AggregationsConstants.ExtendedStats => GetExtendedStatsAggregate(column.ColumnName, columnMetadata, row, logger),
                    AggregationsConstants.TopHits => GetTopHitsAggregate(column.ColumnName, columnMetadata, row, logger, query),
                    _ => throw new InvalidOperationException($"Failed to parse column metadata. {metric} is invalid."),
                };

                aggregateDictionary.Add(key, aggregate);
```

Reply:
Done.